### PR TITLE
areaPlan wrapper

### PR DESCRIPTION
**Pull request by ksobon**
### Purpose

This addresses multiple reports that Area Plans were not being returned properly, either via `Document.ActiveView` nor `ToDSType()` calls. 

image

Specifically this issue: https://github.com/DynamoDS/DynamoRevit/issues/1622

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] Snapshot of UI changes, if any.

### Reviewers

@mjkkirschner @jnealb @riteshchandawar @aparajit-pratap 

### FYIs

I cleaned up the `ViewPlan` class a little bit while I was at it.

### CODE REVIEW

File: test/Libraries/RevitNodesTests/Elements/Views/AreaPlanViewTests.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Linq;
+using NUnit.Framework;
+using RTF.Framework;
+using Revit.Elements.Views;
+using RevitServices.Persistence;
+using RevitTestServices;
+using Revit.Elements;
+
+namespace RevitNodesTests.Elements.Views
```
Comment (Racel) [inline test/Libraries/RevitNodesTests/Elements/Views/AreaPlanViewTests.cs]: Can you check if the properties are equal to the input values? Don't just check that they are not null.

**Comment by Racel** [comment]
The team has reviewed. This looks pretty clean. We just need to confirm if Revit will work with the C#6 features you are using. We will attempt to resolve the merge conflicts, then run the RTF tests.

File: src/Libraries/RevitNodes/Elements/Views/AreaPlanView.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+using RevitServices.Persistence;
+using RevitServices.Transactions;
+using DynamoServices;
+
+namespace Revit.Elements.Views
+{
+    /// <summary>
+    /// A Revit Area Plan View
+    /// </summary>
+    [RegisterForTrace]
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/Views/AreaPlanView.cs]: I'm not sure this actually does anything. Will look into it.

File: src/Libraries/RevitNodes/Elements/Views/FloorPlanView.cs [hunk 3]
```diff
@@ -76,7 +71,7 @@ public static FloorPlanView ByLevel(Level level)
         {
             if (level == null)
             {
-                throw new ArgumentNullException("level");
+                throw new ArgumentNullException(nameof(level));
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/Views/FloorPlanView.cs]: still making a decision around c# 6.0 features.

File: src/Libraries/RevitNodes/RevitNodes.csproj [hunk 4]
```diff
@@ -238,7 +239,6 @@
     <EmbeddedResource Include="Properties\Resources.resx">
       <Generator>ResXFileCodeGenerator</Generator>
       <SubType>Designer</SubType>
-      <LastGenOutput>Resources.Designer.cs</LastGenOutput>
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/RevitNodes.csproj]: this looks incorrect

**Pull request merged**